File: algoDat/public/code/vorlesung-12/vorlagen/java/StringSearchMain.java
/**
 * Copyright 2017, University of Freiburg,
 * Chair of Algorithms and Data Structures.
 * Author: Axel Lehmann <lehmann@cs.uni-freiburg.de>.
 */
import java.nio.file.Files;
import java.nio.file.Paths;
import java.io.IOException;

/**
 * Main function.
 */
public class StringSearchMain {

  /**
   * @param args
   */
  public static void main(String[] args) {
    if (args.length != 2) {
      System.out.println("Usage: java -jar stringsearch.jar <file> <pattern>");
      System.exit(1);
    }
    String fileName = args[0];
    String pattern = args[1];
    StringSearch ss = new StringSearch();
    try {
      for (String line : Files.readAllLines(Paths.get(fileName))) {
        for (int i : ss.findMatchesKmp(line, pattern)) {
          System.out.println(line.substring(i, i + pattern.length()));
        }
      }
    } catch (IOException e) {
      e.printStackTrace();
    }
  }
}

File: algoDat/public/code/vorlesung-12/StringSearchMain.cpp
// Copyright 2015, University of Freiburg,
// Chair of Algorithms and Data Structures.
// Author: Hannah Bast <bast@cs.uni-freiburg.de>.

#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include "./StringSearch.h"

// Main function.
int main(int argc, char** argv) {
  if (argc != 3) {
    std::cout << "Usage: ./StringSearchMain <file> <pattern>" << std::endl;
    exit(1);
  }
  std::string fileName = argv[1];
  std::string pattern = argv[2];
  std::ifstream file(fileName.c_str());
  std::string line;
  StringSearch ss;
  while (file >> line) {
    std::vector<size_t> matches = ss.findMatchesKmp(line, pattern);
    for (auto i : matches) {
      std::cout << line.substr(i, pattern.size()) << std::endl;
    }
  }
}
